components/common/usage_percent_bar/index.tsx

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import styled from 'styled-components';
+
+type Thresholds = {
+    ok: number;
+    warn: number;
+    danger: number;
+    exceeded: number;
+}
+
+type Props = {
+    percent: number;
+    thresholds?: Thresholds;
+    barWidth?: number;
+}
+
+const defaultThresholds = {
+    ok: 0,
+    warn: 40,
+    danger: 80,
+    exceeded: 100.0000001,
+};
+
+type BarBackgroundProps = {
+    width: number;
+    thresholds: Thresholds;
+    percent: number;
+}
+
+const BarBackground = styled.div<BarBackgroundProps>`
+    height: ${(props) => Math.ceil(props.width / 20)}px;
+    width: ${(props) => props.width}px;
+    background: ${(props) => (
+        isExceeded(props.percent, props.thresholds) ?
+            'var(--dnd-indicator)' :
+            'rgba(var(--center-channel-text-rgb), 0.11)'
+    )};
+    border-radius: 8px;
+    position: relative;
+`;
+
+type BarForegroundProps = {
+    width: number;
+    percent: number;
+    thresholds: Thresholds;
+}
+
+function getColor(percent: number, thresholds: Thresholds): string {
+    if (percent >= thresholds.ok && percent < thresholds.warn) {
```

Comment:
nit: always when I see more than one if-else I wonder if is the time for a switch